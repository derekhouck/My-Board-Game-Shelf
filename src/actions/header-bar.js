export const TOGGLE_NAV_LIST = 'TOGGLE_NAV_LIST';
export const toggleNavList = () => ({
  type: TOGGLE_NAV_LIST
});