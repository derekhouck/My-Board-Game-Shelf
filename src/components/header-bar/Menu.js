import Menu from 'react-burger-menu/lib/menus/slide';
import {decorator as reduxBurgerMenu} from 'redux-burger-menu';
import './Menu.css';

export default reduxBurgerMenu(Menu);