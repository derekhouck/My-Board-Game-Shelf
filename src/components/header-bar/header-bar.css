.bm-item.admin-menu {
  border-top: 2px solid #19000b;
  padding-top: 1em;
  margin-top: 1em;
}