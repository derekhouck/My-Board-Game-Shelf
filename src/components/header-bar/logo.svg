<?xml version="1.0" encoding="UTF-8"?>
<svg width="669px" height="106px" viewBox="0 0 669 106" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 53 (72520) - https://sketchapp.com -->
    <title>mbgs logo</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <path d="M381.438828,85.3523146 C380.677935,91.5492971 378.938579,96.8340166 376.220708,101.206632 C373.502837,105.579247 370.089238,109.074338 365.979809,111.69201 C361.870379,114.309682 357.222169,116.080603 352.035038,117.004827 C346.847907,117.929051 341.382099,118.038484 335.637451,117.33313 C333.421012,117.060985 331.025851,116.468451 328.451896,115.555509 C325.87794,114.642567 323.288149,113.578466 320.682445,112.363176 C318.07674,111.147886 315.581179,109.877261 313.195687,108.551263 C310.810195,107.225265 308.700501,105.990539 306.866542,104.84705 L313.284271,77.2594781 C316.845862,80.0384714 320.893837,82.314695 325.428319,84.0882172 C329.962801,85.8617394 334.491646,87.0261815 339.014991,87.5815785 C339.874427,87.6871039 340.918946,87.7809187 342.14858,87.8630257 C343.378214,87.9451327 344.56795,87.8845977 345.717825,87.6814191 C346.867699,87.4782405 347.86876,87.084615 348.721038,86.5005307 C349.573316,85.9164464 350.077204,84.9911499 350.232715,83.7246133 C350.33824,82.8651778 350.157094,82.0968219 349.689271,81.4195227 C349.221449,80.7422236 348.581433,80.1585778 347.769207,79.6685681 C346.95698,79.1785584 346.020575,78.7651368 344.959965,78.428291 C343.899356,78.0914453 342.854436,77.8139224 341.825174,77.5957142 C340.795913,77.3775061 339.828962,77.2128647 338.924293,77.1017854 L336.753098,76.8351962 C332.863022,76.3575548 329.406866,75.2444717 326.384527,73.4959134 C323.362188,71.7473551 320.85748,69.5228768 318.870328,66.8224118 C316.883176,64.1219468 315.460462,61.0316759 314.602142,57.5515062 C313.743822,54.0713366 313.553485,50.3862687 314.031127,46.4961921 C314.686495,41.1586451 316.348097,36.5071843 319.015982,32.5416704 C321.683867,28.5761564 324.997682,25.3328229 328.957526,22.8115725 C332.91737,20.2903221 337.323943,18.5241675 342.177377,17.5130556 C347.030812,16.5019437 351.945295,16.3018591 356.920974,16.9127957 C359.137413,17.1849402 361.411949,17.5445688 363.744649,17.9916922 C366.07735,18.4388157 368.383434,19.0089341 370.662971,19.7020646 C372.942508,20.395195 375.151585,21.1944547 377.290268,22.0998674 C379.428951,23.0052802 381.435082,24.0551094 383.308721,25.2493865 L377.348077,52.4798447 C374.399943,51.0618054 371.284021,49.7953602 368.000215,48.6804711 C364.71641,47.5655819 361.468768,46.8109827 358.257194,46.4166509 C357.669159,46.3444493 356.910121,46.2627299 355.980059,46.1714903 C355.049996,46.0802506 354.069161,46.0286912 353.037523,46.0168105 C352.005885,46.0049299 350.956013,46.0482018 349.887874,46.1466278 C348.819735,46.2450538 347.854974,46.4365203 346.993563,46.7210332 C346.132152,47.005546 345.413796,47.4338834 344.838472,48.006058 C344.263149,48.5782326 343.945346,49.2968039 343.885054,50.1617934 C343.8081,51.1624833 344.082689,52.0111845 344.708829,52.7079226 C345.334969,53.4046607 346.181308,53.9906825 347.247873,54.4660056 C348.314438,54.9413287 349.506565,55.3287557 350.82429,55.6282981 C352.142014,55.9278406 353.455354,56.1694491 354.764347,56.3531309 C356.07334,56.5368127 357.294625,56.6867677 358.428238,56.8030002 C359.561851,56.9192327 360.484957,57.0670119 361.197585,57.2463423 C364.788491,57.9168271 367.950834,59.085664 370.68471,60.752888 C373.418585,62.4201121 375.658639,64.4628721 377.404937,66.8812293 C379.151235,69.2995865 380.371189,72.0779947 381.064836,75.2165373 C381.758483,78.35508 381.883146,81.7336386 381.438828,85.3523146 Z M462.935547,114.265625 L428.755859,115.90625 L428.482422,87.0585938 L412.759766,87.0585938 L411.802734,117 L381.177734,117 L383.775391,20.8867188 L416.314453,22.390625 L414.263672,71.4726563 L429.439453,71.4726563 L429.576172,20.0664063 L460.748047,20.8867188 L462.935547,114.265625 Z M539.115906,51.9279848 C540.075093,55.775075 540.768954,59.6882573 541.197508,63.6676492 C533.757262,66.0393621 526.435766,68.6984911 519.232799,71.645116 C512.029831,74.5917409 504.824216,77.5272669 497.615736,80.451782 C499.559251,83.3489271 501.995102,85.4422383 504.923361,86.7317783 C507.851621,88.0213183 511.018151,88.2416171 514.423047,87.3926812 C516.191824,86.9516755 518.043707,86.1376921 519.978753,84.9507065 C521.913799,83.7637208 523.796417,82.4136889 525.626664,80.9005701 C527.456912,79.3874513 529.179401,77.8190224 530.794182,76.1952362 C532.408964,74.57145 533.785971,73.1243849 534.925246,71.8539976 L539.249547,102.761083 C537.855743,104.517647 536.268637,106.11103 534.488181,107.541282 C532.707725,108.971534 530.805813,110.244183 528.782388,111.359268 C526.758963,112.474352 524.674872,113.440167 522.530051,114.25674 C520.385231,115.073314 518.251586,115.746194 516.129053,116.275401 C511.75133,117.36689 507.600695,117.767697 503.677023,117.477834 C499.753351,117.187971 496.053915,116.337317 492.578604,114.925847 C489.103292,113.514377 485.89637,111.625057 482.957739,109.257832 C480.019108,106.890607 477.390231,104.164393 475.071029,101.079107 C472.751827,97.9938212 470.736797,94.6213976 469.02588,90.9617348 C467.314962,87.302072 465.968905,83.5045331 464.987667,79.569004 C463.962329,75.4565973 463.312866,71.2505399 463.039258,66.9507058 C462.76565,62.6508716 462.917548,58.4093991 463.494956,54.226161 C464.072364,50.0429228 465.116638,46.0368899 466.62781,42.2079421 C468.138982,38.3789943 470.139027,34.8626589 472.628004,31.6588305 C475.116981,28.455002 478.147244,25.6798658 481.718884,23.3333386 C485.290524,20.9868113 489.442021,19.223729 494.1735,18.0440389 C498.109029,17.0628013 501.836859,16.7087018 505.357103,16.9817299 C508.877347,17.254758 512.148525,18.03606 515.170736,19.3256595 C518.192947,20.6152589 520.974428,22.3523312 523.515262,24.5369283 C526.056096,26.7215254 528.328645,29.243035 530.332977,32.1015327 C532.337309,34.9600303 534.076179,38.072535 535.54964,41.4391401 C537.023101,44.8057451 538.211844,48.3019917 539.115906,51.9279848 Z M515.525012,54.1871516 C515.216308,52.9490076 514.766684,51.7107931 514.176126,50.4724709 C513.585568,49.2341486 512.859517,48.1587902 511.997951,47.2463633 C511.136385,46.3339365 510.133759,45.6563092 508.990042,45.2134612 C507.846325,44.7706132 506.544865,44.7311046 505.085624,45.0949343 C503.316847,45.5359399 501.876654,46.3999221 500.765002,47.6869066 C499.65335,48.9738911 498.790282,50.4572033 498.175772,52.1368878 C497.561261,53.8165723 497.153954,55.5737333 496.953837,57.4084236 C496.753721,59.2431138 496.675298,60.9065328 496.718566,62.3987304 L515.88884,55.6463854 L515.525012,54.1871516 Z M602.714423,87.8164028 L597.733105,117.893753 L540,118.379397 L548.991602,21.1525923 L584.451887,23.6322169 L575.028124,85.8803882 L602.714423,87.8164028 Z M669.214844,21 C668.941405,25.2838756 668.690756,29.5221145 668.462891,33.7148438 C668.235025,37.907573 667.938804,42.1458119 667.574219,46.4296875 L638.589844,47.9335938 L638.589844,56.546875 L663.746094,56.546875 L662.378906,77.7382813 L638.453125,78.421875 L638.589844,116.15625 L604,116.839844 L606.1875,21 L669.214844,21 Z" id="path-1"></path>
    </defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="Group" transform="translate(-1.000000, -13.000000)">
            <text id="My" fill="#E18258" font-family="LuckiestGuy-Regular, Luckiest Guy" font-size="72" font-weight="normal">
                <tspan x="0" y="117">My</tspan>
            </text>
            <text id="Board" fill="#E18258" font-family="LuckiestGuy-Regular, Luckiest Guy" font-size="72" font-weight="normal">
                <tspan x="95" y="65">Board</tspan>
            </text>
            <text id="Game" fill="#E18258" font-family="LuckiestGuy-Regular, Luckiest Guy" font-size="72" font-weight="normal">
                <tspan x="109" y="117">Game</tspan>
            </text>
            <g id="Shelf" fill-rule="nonzero">
                <use fill="#000000" xlink:href="#path-1"></use>
                <use fill="#E18258" xlink:href="#path-1"></use>
            </g>
        </g>
    </g>
</svg>