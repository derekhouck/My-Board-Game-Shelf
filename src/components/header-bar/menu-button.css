.menu-button {
  color: #d19cb8;
  text-align: center;
}

.menu-button__box {
  display: inline-block;
  background-color: #4f2119;
  color: #f9fafa;
  border: 4px solid #d19cb8;
  padding: 0 4px;
}

.menu-button__line {
  width: 24px;
  height: 4px;
  background-color: #d19cb8;
  margin: 6px auto;
  transition: 0.4s;
}