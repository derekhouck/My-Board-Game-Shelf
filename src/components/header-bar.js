import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { clearAuth } from '../actions/auth';
import { clearAuthToken } from '../local-storage';

export class HeaderBar extends React.Component {
  logOut() {
    this.props.dispatch(clearAuth());
    clearAuthToken();
  }

  render() {
    // Only render the log out button if we are logged in
    let appTitle, headerNav;
    if (this.props.loggedIn) {
      appTitle = (
        <Link to="/dashboard" className="App-title">
          My Board Game Shelf
          </Link>
      );
      headerNav = (
        <nav className="App-nav">
          <ul className="App-nav__list">
            <li>
              <button onClick={() => this.logOut()}>Log out</button>
            </li>
            <li>
              <Link to="/account">Your Account</Link>
            </li>
          </ul>
        </nav>
      );
    } else {
      appTitle = (
        <div className="App-title">My Board Game Shelf</div>
      );
    }

    return (
      <header className="App-header">
        {appTitle}
        {headerNav}
      </header>
    );
  }
}

const mapStateToProps = state => ({
  loggedIn: state.auth.currentUser !== null
});

export default connect(mapStateToProps)(HeaderBar);