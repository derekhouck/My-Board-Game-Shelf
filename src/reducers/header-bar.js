import { TOGGLE_NAV_LIST } from '../actions/header-bar';

const initialState = {
  expanded: false
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case TOGGLE_NAV_LIST:
      return Object.assign({}, state, {
        expanded: !state.expanded
      });
    default:
      return state;
  }
}