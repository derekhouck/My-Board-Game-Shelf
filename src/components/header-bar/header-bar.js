import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { clearAuth } from '../../actions/auth';
import { clearAuthToken } from '../../local-storage';
import { action as toggleMenu } from 'redux-burger-menu';
import logo from './logo.svg';
import './header-bar.css';

import Menu from './Menu';
import MenuButton from './menu-button';

export class HeaderBar extends React.Component {
  logOut() {
    this.props.dispatch(toggleMenu(false));
    this.props.dispatch(clearAuth());
    clearAuthToken();
  }

  render() {
    const { currentUser, dispatch, loggedIn } = this.props;
    let headerNav;
    const adminMenu = (
      <div className="admin-menu">
        <Link to="/users" onClick={() => dispatch(toggleMenu(false))}>Users</Link>
        <Link to="/admin/games" onClick={() => dispatch(toggleMenu(false))}>Games</Link>
      </div>
    );
    const appTitle = (
      <Link to={loggedIn ? '/dashboard' : '/'} className="App-title">
        <img src={logo} alt="My Board Game Shelf" />
      </Link>
    );

    // Only render the log out button if we are logged in
    if (loggedIn) {
      headerNav = (
        <Menu
          right
          pageWrapId={"AppMain"}
          outerContainerId={"App"}
          customBurgerIcon={<MenuButton />}
        >
          <Link
            to="/games/add"
            onClick={() => dispatch(toggleMenu(false))}
          >
            Add a game
          </Link>
          <Link to="/account" onClick={() => dispatch(toggleMenu(false))}>Your account</Link>
          <button className="App-header__logout" onClick={() => this.logOut()}>Log out</button>
          {currentUser && currentUser.admin ? adminMenu : null}
        </Menu>
      );
    } else {
      headerNav = (<Link to="/sign-in">Sign in</Link>);
    }

    return (
      <header className="App-header">
        {appTitle}
        {headerNav}
      </header>
    );
  }
}

const mapStateToProps = state => ({
  currentUser: state.auth.currentUser,
  loggedIn: state.auth.currentUser !== null
});

export default connect(mapStateToProps)(HeaderBar);