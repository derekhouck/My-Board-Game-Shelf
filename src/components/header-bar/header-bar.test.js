import React from 'react';
import { shallow } from 'enzyme';
import { HeaderBar } from './header-bar';

describe('<HeaderBar />', function () {
  it('renders without crashing', () => {
    const wrapper = shallow(<HeaderBar />);
    expect(wrapper.find('.App-header')).toHaveLength(1);
  });

  it('dispatches logout action', function () {
    const callback = jest.fn();
    const wrapper = shallow(<HeaderBar loggedIn={true} dispatch={callback} />);
    wrapper.find('.App-header__logout').simulate('click');
    expect(callback).toHaveBeenCalledTimes(2);
  });
});
