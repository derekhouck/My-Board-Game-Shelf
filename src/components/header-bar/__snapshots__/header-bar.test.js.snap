// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderBar /> renders without crashing 1`] = `ShallowWrapper {}`;
